src/Domain/LeanCode.DomainModels.EF/EFRepository.cs

```diff
@@ -58,4 +60,28 @@ public virtual void Update(TEntity entity)
     }
 
     public abstract Task<TEntity?> FindAsync(TIdentity id, CancellationToken cancellationToken = default);
+
+    [SuppressMessage(
+        "?",
+        "EF1001",
+        Justification = "There basically resembles the EntityFinder.FindTracked to mimic `FindAsync` snapshot re-use."
+    )]
+    protected TEntity? FindTracked(TIdentity id)
+    {
+        // Safety: aggregates are bound to have Id as a primary key by design.
+        var primaryKey = DbContext.Model.FindEntityType(typeof(TEntity))!.FindPrimaryKey()!;
```

Comment:
This wouldn't work for compound keys, would it?

Reply:
As far as I know, there is still a single PK in the model, it just has multiple properties. But I'll add a test for that.

Reply:
But `TIdentity id` can't represent multiple properties outside of some by-convention `(FooId FooId, BarId BarId)` which EFCore is unlikely to understand.

Reply:
Yeah, forgot about the "magic" that maps tuple to composite keys and that EF interprets that as an array. :(